image_view/src/nodelets/image_nodelet.cpp

```diff
@@ -167,23 +180,57 @@ void ImageNodelet::onInit()
   image_transport::ImageTransport it(nh);
   image_transport::TransportHints hints(transport, ros::TransportHints(), getPrivateNodeHandle());
   sub_ = it.subscribe(topic, 1, &ImageNodelet::imageCb, this, hints);
+  pub_ = local_nh.advertise<sensor_msgs::Image>("output", 1);
+
+  dynamic_reconfigure::Server<image_view::ImageViewConfig>::CallbackType f =
+    boost::bind(&ImageNodelet::reconfigureCb, this, _1, _2);
+  srv_.setCallback(f);
+}
+
+void ImageNodelet::reconfigureCb(image_view::ImageViewConfig &config, uint32_t level)
+{
+  do_dynamic_scaling_ = config.do_dynamic_scaling;
+  colormap_ = config.colormap;
+  min_image_value_ = config.min_image_value;
+  max_image_value_ = config.max_image_value;
 }
 
 void ImageNodelet::imageCb(const sensor_msgs::ImageConstPtr& msg)
 {
   // We want to scale floating point images so that they display nicely
-  bool do_dynamic_scaling = (msg->encoding.find("F") != std::string::npos);
+  bool do_dynamic_scaling = do_dynamic_scaling_ & (msg->encoding.find("F") != std::string::npos);
```

Comment:
Do you intend to do a bitwise operation, or do you actually mean `&&`?

Reply:
If the message is asking for scaling because its a float image, shouldn't we always do it? Maybe the logic should be that it does scaling for floats, but then it checks the `do_dynamic_scaling_` for non-float images to do scalling on those?